lib/oli/authoring/course.ex

```diff
@@ -576,4 +578,34 @@ defmodule Oli.Authoring.Course do
     |> Oli.Analytics.DatashopExportWorker.new()
     |> Oban.insert()
   end
+
+  @cc_options Oli.Authoring.Course.CreativeCommons.cc_options() |> Enum.map(& &1.id)
+
+  @type license_types ::
+          :none | :cc_by | :cc_by_sa | :cc_by_nd | :cc_by_nc | :cc_by_nc_sa | :cc_by_nc_nd
+  @spec get_project_license(integer()) ::
+          %{license_type: :custom, custom_license_details: String.t()}
+          | %{license_type: license_types, custom_license_details: <<>>}
+          | nil
+  def get_project_license(revision_id) do
+    from(pr in PublishedResource,
+      join: spp in SectionsProjectsPublications,
+      on: pr.publication_id == spp.publication_id,
+      join: p in assoc(spp, :project),
+      where: pr.revision_id == ^revision_id,
+      distinct: true,
+      select: p.attributes
+    )
+    |> Repo.one()
+    |> case do
+      %{license: %{license_type: :custom} = license} ->
+        Map.from_struct(license)
+
+      %{license: %{license_type: license_type} = license} when license_type in @cc_options ->
+        Map.from_struct(license)
```

Comment:
Is it necessary to distinguish these two cases when the output is the same? What do you think about something like:

```elixir
|> case do
      %{license: %{license_type: license_type} = license} when license_type in [:custom | @cc_options] ->
        Map.from_struct(license)
        
      nil ->
        nil
    end
```

Reply:
Nice catch! We don't even need the guard because the `@cc_options` includes `:custom`.